<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>clustOpt: Quickstart Guide • clustOpt</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="clustOpt: Quickstart Guide">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-primary" data-bs-theme="dark" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">clustOpt</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="Released version">1.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../index.html" aria-label="Home"><span class="fa fas fa-home fa-lg"></span></a></li>
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="../articles/quick_start_guide.html">Quick Start Guide</a></li>
  </ul>
</li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/gladstone-institutes/clustOpt" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>clustOpt: Quickstart Guide</h1>
            
            <h4 data-toc-skip class="date">Compiled: 2025-07-23</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/gladstone-institutes/clustOpt/blob/HEAD/vignettes/quick_start_guide.Rmd" class="external-link"><code>vignettes/quick_start_guide.Rmd</code></a></small>
      <div class="d-none name"><code>quick_start_guide.Rmd</code></div>
    </div>

    
    
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://satijalab.org/seurat" class="external-link">Seurat</a></span><span class="op">)</span></span>
<span><span class="co"># Only v5 Seurat Assays are supported</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/options.html" class="external-link">options</a></span><span class="op">(</span>Seurat.object.assay.version <span class="op">=</span> <span class="st">"v5"</span><span class="op">)</span></span>
<span><span class="co"># Increase future globals limit for large datasets</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/options.html" class="external-link">options</a></span><span class="op">(</span>future.globals.maxSize <span class="op">=</span> <span class="fl">2e9</span><span class="op">)</span>  <span class="co"># 2GB limit</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://tidyr.tidyverse.org" class="external-link">tidyr</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://gladstone-institutes.github.io/clustOpt">clustOpt</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://future.futureverse.org" class="external-link">future</a></span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">42</span><span class="op">)</span></span></code></pre></div>
<p>Read in the tutorial dataset. This data was generated from the <a href="https://doi.org/10.1101/2024.06.30.601119" class="external-link">Asian Immune Diversity
Atlas</a> (data freeze v1). It contains only B cells, Monocytes, and NK
cells from 10 donors which have been sketched to a total of 1000 cells
using the leverage score based method of Seurat’s SketchData
function.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">input</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">readRDS</a></span><span class="op">(</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span></span>
<span>    <span class="st">"extdata"</span>,</span>
<span>    <span class="st">"1000_cell_sketch_10_donors_3_celltypes_AIDA.rds"</span>,</span>
<span>    package <span class="op">=</span> <span class="st">"clustOpt"</span></span>
<span>  <span class="op">)</span></span>
<span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">input</span><span class="op">@</span><span class="va">meta.data</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/summarise.html" class="external-link">summarize</a></span><span class="op">(</span></span>
<span>    Donors <span class="op">=</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/n_distinct.html" class="external-link">n_distinct</a></span><span class="op">(</span><span class="va">donor_id</span><span class="op">)</span>,</span>
<span>    Celltypes <span class="op">=</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/n_distinct.html" class="external-link">n_distinct</a></span><span class="op">(</span><span class="va">broad_cell_type</span><span class="op">)</span>,</span>
<span>    `Cell Subtypes` <span class="op">=</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/n_distinct.html" class="external-link">n_distinct</a></span><span class="op">(</span><span class="va">author_cell_type</span><span class="op">)</span>,</span>
<span>    Ethnicities <span class="op">=</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/n_distinct.html" class="external-link">n_distinct</a></span><span class="op">(</span><span class="va">Ethnicity_Selfreported</span><span class="op">)</span>,</span>
<span>    Countries <span class="op">=</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/n_distinct.html" class="external-link">n_distinct</a></span><span class="op">(</span><span class="va">Country</span><span class="op">)</span></span>
<span>  <span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="https://tidyr.tidyverse.org/reference/pivot_longer.html" class="external-link">pivot_longer</a></span><span class="op">(</span>cols <span class="op">=</span> <span class="fu"><a href="https://tidyselect.r-lib.org/reference/everything.html" class="external-link">everything</a></span><span class="op">(</span><span class="op">)</span>, names_to <span class="op">=</span> <span class="st">"Metric"</span>, values_to <span class="op">=</span> <span class="st">"Count"</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 5 × 2</span></span></span>
<span><span class="co">#&gt;   Metric        Count</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>         <span style="color: #949494; font-style: italic;">&lt;int&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> Donors           10</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> Celltypes         3</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> Cell Subtypes    10</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span> Ethnicities       5</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">5</span> Countries         3</span></span></code></pre></div>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">input</span> <span class="op">&lt;-</span> <span class="va">input</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="https://satijalab.org/seurat/reference/SCTransform.html" class="external-link">SCTransform</a></span><span class="op">(</span>verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="https://satijalab.org/seurat/reference/RunPCA.html" class="external-link">RunPCA</a></span><span class="op">(</span>verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://satijalab.org/seurat/reference/ElbowPlot.html" class="external-link">ElbowPlot</a></span><span class="op">(</span><span class="va">input</span>, ndims <span class="op">=</span> <span class="fl">50</span><span class="op">)</span></span></code></pre></div>
<p><img src="quick_start_guide_files/figure-html/elbowplot-1.png" width="768"></p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">input</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/RunUMAP.html" class="external-link">RunUMAP</a></span><span class="op">(</span><span class="va">input</span>, dims <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">20</span>, verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://satijalab.org/seurat/reference/DimPlot.html" class="external-link">DimPlot</a></span><span class="op">(</span><span class="va">input</span>,</span>
<span>  group.by <span class="op">=</span> <span class="st">"donor_id"</span>,</span>
<span>  pt.size <span class="op">=</span> <span class="fl">.8</span></span>
<span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/coord_fixed.html" class="external-link">coord_fixed</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<p><img src="quick_start_guide_files/figure-html/UMAP-1.png" width="768"></p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://satijalab.org/seurat/reference/DimPlot.html" class="external-link">DimPlot</a></span><span class="op">(</span><span class="va">input</span>,</span>
<span>  group.by <span class="op">=</span> <span class="st">"broad_cell_type"</span>,</span>
<span>  pt.size <span class="op">=</span> <span class="fl">.8</span></span>
<span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/coord_fixed.html" class="external-link">coord_fixed</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<p><img src="quick_start_guide_files/figure-html/unnamed-chunk-2-1.png" width="768"></p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://satijalab.org/seurat/reference/DimPlot.html" class="external-link">DimPlot</a></span><span class="op">(</span><span class="va">input</span>,</span>
<span>  group.by <span class="op">=</span> <span class="st">"author_cell_type"</span>,</span>
<span>  pt.size <span class="op">=</span> <span class="fl">.8</span></span>
<span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/coord_fixed.html" class="external-link">coord_fixed</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<p><img src="quick_start_guide_files/figure-html/unnamed-chunk-3-1.png" width="768"></p>
<p>Subject-wise cross validation is parallelized with
<code>future</code> and <code>future.batchtools</code> can be used for
HPC job schedulers. For details on which plan to use for your setup, see
the future package documentation. The code below requires a large amount
of RAM (32 GB recommended) and 10 cores to run in about 20 minutes. The
repeated messages about “SeuratObject” are a side effect of using the
future plan “multisession” and should not occur with other future
plans.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://future.futureverse.org/reference/plan.html" class="external-link">plan</a></span><span class="op">(</span><span class="st">"multisession"</span>, workers <span class="op">=</span> <span class="fl">10</span><span class="op">)</span></span>
<span></span>
<span><span class="va">sil_dist</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/clust_opt.html">clust_opt</a></span><span class="op">(</span><span class="va">input</span>,</span>
<span>  subject_ids <span class="op">=</span> <span class="st">"donor_id"</span>,</span>
<span>  ndim <span class="op">=</span> <span class="fl">20</span>,</span>
<span>  verbose <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  train_with <span class="op">=</span> <span class="st">"even"</span></span>
<span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">progressr</span><span class="fu">::</span><span class="fu"><a href="https://progressr.futureverse.org/reference/with_progress.html" class="external-link">with_progress</a></span><span class="op">(</span><span class="op">)</span> <span class="co"># Optional, just adds progress bar</span></span>
<span><span class="co">#&gt; Input is small enough to run with all cells</span></span>
<span><span class="co">#&gt; Using 10 subject(s) that have at least 50 cells:</span></span>
<span><span class="co">#&gt; JP_RIK_H073 (100 cells)</span></span>
<span><span class="co">#&gt; JP_RIK_H131 (83 cells)</span></span>
<span><span class="co">#&gt; KR_SGI_H002 (76 cells)</span></span>
<span><span class="co">#&gt; KR_SGI_H136 (127 cells)</span></span>
<span><span class="co">#&gt; SG_HEL_H09a (52 cells)</span></span>
<span><span class="co">#&gt; SG_HEL_H077 (147 cells)</span></span>
<span><span class="co">#&gt; SG_HEL_H138 (130 cells)</span></span>
<span><span class="co">#&gt; SG_HEL_H153 (85 cells)</span></span>
<span><span class="co">#&gt; SG_HEL_H166 (123 cells)</span></span>
<span><span class="co">#&gt; SG_HEL_H319 (77 cells)</span></span>
<span><span class="co">#&gt; Found 110 combinations of test subject and resolution</span></span>
<span><span class="co">#&gt; Holdout subject: JP_RIK_H073</span></span>
<span><span class="co">#&gt; Loading required package: SeuratObject</span></span>
<span><span class="co">#&gt; Loading required package: sp</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Attaching package: ‘SeuratObject’</span></span>
<span><span class="co">#&gt; The following objects are masked from ‘package:base’:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     intersect, t</span></span>
<span><span class="co">#&gt; Loading required package: SeuratObject</span></span>
<span><span class="co">#&gt; Loading required package: sp</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Attaching package: ‘SeuratObject’</span></span>
<span><span class="co">#&gt; The following objects are masked from ‘package:base’:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     intersect, t</span></span>
<span><span class="co">#&gt; Loading required package: SeuratObject</span></span>
<span><span class="co">#&gt; Loading required package: sp</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Attaching package: ‘SeuratObject’</span></span>
<span><span class="co">#&gt; The following objects are masked from ‘package:base’:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     intersect, t</span></span>
<span><span class="co">#&gt; Loading required package: SeuratObject</span></span>
<span><span class="co">#&gt; Loading required package: sp</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Attaching package: ‘SeuratObject’</span></span>
<span><span class="co">#&gt; The following objects are masked from ‘package:base’:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     intersect, t</span></span>
<span><span class="co">#&gt; Loading required package: SeuratObject</span></span>
<span><span class="co">#&gt; Loading required package: sp</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Attaching package: ‘SeuratObject’</span></span>
<span><span class="co">#&gt; The following objects are masked from ‘package:base’:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     intersect, t</span></span>
<span><span class="co">#&gt; Loading required package: SeuratObject</span></span>
<span><span class="co">#&gt; Loading required package: sp</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Attaching package: ‘SeuratObject’</span></span>
<span><span class="co">#&gt; The following objects are masked from ‘package:base’:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     intersect, t</span></span>
<span><span class="co">#&gt; Loading required package: SeuratObject</span></span>
<span><span class="co">#&gt; Loading required package: sp</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Attaching package: ‘SeuratObject’</span></span>
<span><span class="co">#&gt; The following objects are masked from ‘package:base’:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     intersect, t</span></span>
<span><span class="co">#&gt; Loading required package: SeuratObject</span></span>
<span><span class="co">#&gt; Loading required package: sp</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Attaching package: ‘SeuratObject’</span></span>
<span><span class="co">#&gt; The following objects are masked from ‘package:base’:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     intersect, t</span></span>
<span><span class="co">#&gt; Loading required package: SeuratObject</span></span>
<span><span class="co">#&gt; Loading required package: sp</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Attaching package: ‘SeuratObject’</span></span>
<span><span class="co">#&gt; The following objects are masked from ‘package:base’:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     intersect, t</span></span>
<span><span class="co">#&gt; Loading required package: SeuratObject</span></span>
<span><span class="co">#&gt; Loading required package: sp</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Attaching package: ‘SeuratObject’</span></span>
<span><span class="co">#&gt; The following objects are masked from ‘package:base’:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     intersect, t</span></span>
<span><span class="co">#&gt; Loading required package: SeuratObject</span></span>
<span><span class="co">#&gt; Loading required package: sp</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Attaching package: ‘SeuratObject’</span></span>
<span><span class="co">#&gt; The following objects are masked from ‘package:base’:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     intersect, t</span></span>
<span><span class="co">#&gt; Loading required package: SeuratObject</span></span>
<span><span class="co">#&gt; Loading required package: sp</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Attaching package: ‘SeuratObject’</span></span>
<span><span class="co">#&gt; The following objects are masked from ‘package:base’:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     intersect, t</span></span>
<span><span class="co">#&gt; Loading required package: SeuratObject</span></span>
<span><span class="co">#&gt; Loading required package: sp</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Attaching package: ‘SeuratObject’</span></span>
<span><span class="co">#&gt; The following objects are masked from ‘package:base’:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     intersect, t</span></span>
<span><span class="co">#&gt; Loading required package: SeuratObject</span></span>
<span><span class="co">#&gt; Loading required package: sp</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Attaching package: ‘SeuratObject’</span></span>
<span><span class="co">#&gt; The following objects are masked from ‘package:base’:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     intersect, t</span></span>
<span><span class="co">#&gt; Loading required package: SeuratObject</span></span>
<span><span class="co">#&gt; Loading required package: sp</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Attaching package: ‘SeuratObject’</span></span>
<span><span class="co">#&gt; The following objects are masked from ‘package:base’:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     intersect, t</span></span>
<span><span class="co">#&gt; Loading required package: SeuratObject</span></span>
<span><span class="co">#&gt; Loading required package: sp</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Attaching package: ‘SeuratObject’</span></span>
<span><span class="co">#&gt; The following objects are masked from ‘package:base’:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     intersect, t</span></span>
<span><span class="co">#&gt; Loading required package: SeuratObject</span></span>
<span><span class="co">#&gt; Loading required package: sp</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Attaching package: ‘SeuratObject’</span></span>
<span><span class="co">#&gt; The following objects are masked from ‘package:base’:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     intersect, t</span></span>
<span><span class="co">#&gt; Loading required package: SeuratObject</span></span>
<span><span class="co">#&gt; Loading required package: sp</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Attaching package: ‘SeuratObject’</span></span>
<span><span class="co">#&gt; The following objects are masked from ‘package:base’:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     intersect, t</span></span>
<span><span class="co">#&gt; Loading required package: SeuratObject</span></span>
<span><span class="co">#&gt; Loading required package: sp</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Attaching package: ‘SeuratObject’</span></span>
<span><span class="co">#&gt; The following objects are masked from ‘package:base’:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     intersect, t</span></span>
<span><span class="co">#&gt; Loading required package: SeuratObject</span></span>
<span><span class="co">#&gt; Loading required package: sp</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Attaching package: ‘SeuratObject’</span></span>
<span><span class="co">#&gt; The following objects are masked from ‘package:base’:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     intersect, t</span></span>
<span><span class="co">#&gt; Found 2932 (97.73%) shared genes used for projecting test data</span></span>
<span><span class="co">#&gt; Holdout subject: JP_RIK_H131</span></span>
<span><span class="co">#&gt; Found 2923 (97.43%) shared genes used for projecting test data</span></span>
<span><span class="co">#&gt; Holdout subject: KR_SGI_H002</span></span>
<span><span class="co">#&gt; Found 2923 (97.43%) shared genes used for projecting test data</span></span>
<span><span class="co">#&gt; Holdout subject: KR_SGI_H136</span></span>
<span><span class="co">#&gt; Found 2942 (98.07%) shared genes used for projecting test data</span></span>
<span><span class="co">#&gt; Holdout subject: SG_HEL_H09a</span></span>
<span><span class="co">#&gt; Found 2820 (94.00%) shared genes used for projecting test data</span></span>
<span><span class="co">#&gt; Holdout subject: SG_HEL_H077</span></span>
<span><span class="co">#&gt; Found 2988 (99.60%) shared genes used for projecting test data</span></span>
<span><span class="co">#&gt; Holdout subject: SG_HEL_H138</span></span>
<span><span class="co">#&gt; Found 2956 (98.53%) shared genes used for projecting test data</span></span>
<span><span class="co">#&gt; Holdout subject: SG_HEL_H153</span></span>
<span><span class="co">#&gt; Found 2922 (97.40%) shared genes used for projecting test data</span></span>
<span><span class="co">#&gt; Holdout subject: SG_HEL_H166</span></span>
<span><span class="co">#&gt; Found 2956 (98.53%) shared genes used for projecting test data</span></span>
<span><span class="co">#&gt; Holdout subject: SG_HEL_H319</span></span>
<span><span class="co">#&gt; Found 2896 (96.53%) shared genes used for projecting test data</span></span>
<span></span>
<span><span class="co"># Shut down stray workers</span></span>
<span><span class="fu"><a href="https://future.futureverse.org/reference/plan.html" class="external-link">plan</a></span><span class="op">(</span><span class="st">"sequential"</span><span class="op">)</span></span></code></pre></div>
<div class="section level2">
<h2 id="plotting-the-silhouette-score-distributions">Plotting the silhouette score distributions<a class="anchor" aria-label="anchor" href="#plotting-the-silhouette-score-distributions"></a>
</h2>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">plots</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/create_sil_plots.html">create_sil_plots</a></span><span class="op">(</span><span class="va">sil_dist</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://tidyr.tidyverse.org/reference/drop_na.html" class="external-link">drop_na</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">plots</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span></span></code></pre></div>
<p><img src="quick_start_guide_files/figure-html/plotting-1.png" width="768"></p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">plots</span><span class="op">[[</span><span class="fl">2</span><span class="op">]</span><span class="op">]</span></span></code></pre></div>
<p><img src="quick_start_guide_files/figure-html/plotting-2.png" width="768"></p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">plots</span><span class="op">[[</span><span class="fl">3</span><span class="op">]</span><span class="op">]</span></span></code></pre></div>
<p><img src="quick_start_guide_files/figure-html/plotting-3.png" width="768"></p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">plots</span><span class="op">[[</span><span class="fl">4</span><span class="op">]</span><span class="op">]</span></span></code></pre></div>
<p><img src="quick_start_guide_files/figure-html/plotting-4.png" width="768"></p>
<p>The median silhouette scores across all cells start to fall off at
0.2, indicating that this clustering resolution is the most reproducible
across all biological replicates. It is also possible to see a local
maximum which would indicate a reproducible resolution.</p>
</div>
<div class="section level2">
<h2 id="comparison-to-the-cell-annotations">Comparison to the cell annotations<a class="anchor" aria-label="anchor" href="#comparison-to-the-cell-annotations"></a>
</h2>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">input</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/FindNeighbors.html" class="external-link">FindNeighbors</a></span><span class="op">(</span><span class="va">input</span>, dims <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">20</span>, verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="va">input</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/FindClusters.html" class="external-link">FindClusters</a></span><span class="op">(</span><span class="va">input</span>, resolution <span class="op">=</span> <span class="fl">0.2</span><span class="op">)</span></span>
<span><span class="co">#&gt; Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Number of nodes: 1000</span></span>
<span><span class="co">#&gt; Number of edges: 31348</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Running Louvain algorithm...</span></span>
<span><span class="co">#&gt; Maximum modularity in 10 random starts: 0.9287</span></span>
<span><span class="co">#&gt; Number of communities: 4</span></span>
<span><span class="co">#&gt; Elapsed time: 0 seconds</span></span>
<span></span>
<span><span class="fu"><a href="https://satijalab.org/seurat/reference/DimPlot.html" class="external-link">DimPlot</a></span><span class="op">(</span><span class="va">input</span>,</span>
<span>  group.by <span class="op">=</span> <span class="st">"seurat_clusters"</span>,</span>
<span>  pt.size <span class="op">=</span> <span class="fl">.8</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="quick_start_guide_files/figure-html/cluster-1.png" width="768"></p>
<p>We can see that at the reproducible resolution there are 4 clusters,
with the naive B cells being distinct enough to be separated into their
own cluster based on the ability to detect them across the CV folds. In
other words the naive B cells are present in high enough amounts across
the subjects and transcriptomically distinct enough to be consistently
detected.</p>
<p>For convenience we provide a function to calculate the adjusted rand
index for any 2 metadata columns in a seurat object so that the clustOpt
resolution parameter can be compared to clusters derived from other
methods.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/adjusted_rand_index.html">adjusted_rand_index</a></span><span class="op">(</span><span class="va">input</span>,</span>
<span>  meta1 <span class="op">=</span> <span class="st">"seurat_clusters"</span>,</span>
<span>  meta2 <span class="op">=</span> <span class="st">"broad_cell_type"</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 0.9765531</span></span>
<span><span class="fu"><a href="../reference/adjusted_rand_index.html">adjusted_rand_index</a></span><span class="op">(</span><span class="va">input</span>,</span>
<span>  meta1 <span class="op">=</span> <span class="st">"seurat_clusters"</span>,</span>
<span>  meta2 <span class="op">=</span> <span class="st">"author_cell_type"</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 0.8265365</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="larger-datasets">Larger Datasets<a class="anchor" aria-label="anchor" href="#larger-datasets"></a>
</h2>
<p>Due to the resource requirements of clustOpt, we recommend running it
in Rscripts with the <code>future</code> plan set to “multicore” instead
of “multisession” and if possible using <code>future.batchtools</code>.
Additionally we provide a wrapper around <code>Seurat</code>’s
SketchData using the leverage score method to reduce the size of the
input data to clustOpt.</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">input</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/leverage_sketch.html">leverage_sketch</a></span><span class="op">(</span><span class="va">input</span>, sketch_size <span class="op">=</span> <span class="fl">100</span><span class="op">)</span></span>
<span><span class="co">#&gt; Sketching scRNA data: 1000 -&gt; 100 cells</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Removing previously calculated leverage scores...</span></span>
<span><span class="co">#&gt; Normalizing scRNA-seq data...</span></span>
<span><span class="co">#&gt; Finding variable features for scRNA-seq data...</span></span>
<span><span class="co">#&gt; Using 2000 variable features for sketching</span></span>
<span><span class="co">#&gt; Calcuating Leverage Score</span></span>
<span><span class="co">#&gt; Renaming default assay from sketch to RNA</span></span>
<span><span class="va">input</span></span>
<span><span class="co">#&gt; An object of class Seurat </span></span>
<span><span class="co">#&gt; 13855 features across 100 samples within 1 assay </span></span>
<span><span class="co">#&gt; Active assay: RNA (13855 features, 2000 variable features)</span></span>
<span><span class="co">#&gt;  3 layers present: counts, data, scale.data</span></span></code></pre></div>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Natalie Gill, Min-Gyoung Shin, Ayushi Agrawal, Erwin Perez, Reuben Thomas.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.0.</p>
</div>

    </footer>
</div>





  </body>
</html>
